src/data/actions/grades.js

```diff
@@ -97,6 +110,41 @@ const fetchGrades = (
   }
 );
 
+const formatDate = (serverDate) => {
+  const options = { year: 'numeric', month: 'long', day: 'numeric' };
+  const timeOptions = { hour: '2-digit', minute: '2-digit' };
+  const dt = new Date(serverDate);
+  return `${dt.toLocaleDateString('en-US', options)} at ${dt.toLocaleTimeString('en-US', timeOptions)}`;
+};
+
+// Map from REST response to internal properties
+const formatGradeOverrideForDisplay = historyArray => historyArray.map(item => ({
+  date: formatDate(item.history_date),
+  grader: item.history_user,
+  reason: item.override_reason,
+  adjustedGrade: item.earned_graded_override,
+}));
+
+// Redux action creater wraps web api service and dispatches actions using Redux
+// thunk type. function that returns another function
+// the inner (return) function is called implicitly by redux -thunk layer
+const fetchGradeOverrideHistory = (subsectionId, userId) => (
+  dispatch =>
+    LmsApiService.fetchGradeOverrideHistory(subsectionId, userId)
+      .then(response => response.data)
+      .then((data) => {
+        dispatch(gotGradeOverrideHistory(
+          formatGradeOverrideForDisplay(data.history), data.override.earned_all_override,
+          data.override.possible_all_override, data.override.earned_graded_override,
+          data.override.possible_graded_override,
```

Comment:
I would pass along the subsectionId & userId we loaded this from as well, so that the reducer can keep track of which combinations have been loaded already, saving us from needing to hit the backend & make the user wait each time they switch which grade they're viewing.

Some pieces of the puzzle to doing that: a second argument is also provided to invoked thunks alongside `dispatch`; `getState` enables you to read from state to determine how you want your thunk to behave. E.g. if we wanted to read from a portion of state to determine whether we need to load data, we can get a reference to state by invoking that function. Simple example here: https://github.com/reduxjs/redux-thunk#motivation

Reply:
Future enhancement, probably

Reply:
I have a formal rule about this sort of code review discussion: if you are pushing off work for the future, show me tickets or I know you're lying ;).

If you could drop something in Jira linking back to this discussion then I'm more than happy to mark this as resolved.

Reply:
https://openedx.atlassian.net/browse/EDUCATOR-4492